### ISSUE TITLE
beautify with long-ish variables

### ISSUE THREAD

**Issue by jimmywarting** [beautify-web/js-beautify#954]
Basicly implement what this guy did:
https://github.com/zertosh/beautify-with-words

**Comment by bitwiseman** [comment]
At this point this is out of scope.  The beautifier aims to be a whitespace-only reformatter.

**Comment by jimmywarting** [comment]
Could you not include an option on your website? Reverse engineering is basically only what I'm using your website for

**Comment by bitwiseman** [comment]
See #200.  What you're asking for is not possible under the current implementation.

**Comment by jimmywarting** [comment]
A checkbox on http://jsbeautifier.org that replaces all variable with long-ish word
<img width="609" alt="screen shot 2016-06-13 at 11 33 39" src="https://cloud.githubusercontent.com/assets/1148376/16002974/b78d94d0-315a-11e6-8f8d-18e3ca36d7ef.png">

**Comment by jimmywarting** [comment]
This could come in handy is if you need to reverse engineer some minified code and want to replace variable names as you go. Search and replace doesn't work very well on single letter variables as you'll search for 'a' and replace a bunch of unwanted stuff. Having a tool that creates unique var names with scoping taken into account seems somewhat more useful as you'd be able to properly rename those variables.

**Comment by jimmywarting** [comment]
If the beautifier aims to be a whitespace-only reformatter. then don't include it in the module but include it as an option on you website. Having it replace variable names as default would be terrible idea for code editors

**Comment by bitwiseman** [comment]
That is an interesting idea.  I think it would need to be put under a new section of options called "External tools", but it could work.  Not a huge change, but I don't have the bandwidth for it at this time.  Would you be willing to create a PR for this?

**Comment by jimmywarting** [comment]
I'm quite busy with other stuff.

**Comment by bitwiseman** [comment]
Yeah, I know how that goes.  Well, I'll keep this open and hopefully someone will find the time.

**Comment by jimmywarting** [comment]
Could you take a look at this soon? Wish i had this last week

**Comment by bitwiseman** [comment]
Same answer as before from me.  Same answer as before from you.  😄
